[INVE-285] Using uuid as layer keys

corrigancd opened the pull request:
Fix for - https://sirensolutions.atlassian.net/browse/INVE-285

szydan commented on public/vislib/_map.js:
```diff
@@ -527,11 +517,11 @@ define(function (require) {
-          this._poiLayers[layerName].destroy();
+        if (_.has(this.poiLayers, layerName)) {
+          const layer = this.poiLayers[layerName];
+          this.poiLayers[layerName].destroy();
```
just checking here 
above you are also setting isVisible inside similar if 
Is it necessary here too ?

corrigancd commented:
Refactoring and fix for - https://sirensolutions.atlassian.net/browse/INVE-285
To be merged with - https://github.com/sirensolutions/kibi-internal/pull/11685

No change to Kibi-distribution is required as there are no map overlays in the vis objects to add an id field to. 

**POI popup fix**
I fixed the poiPopup issue as part of this refactoring subtask (I initially thought this was the Labelling PAX issue they described). Now if there are too many poi popup fields to fit on the screen: 

1) the popup still appears near the marker 
2) the popup now appears on the screen at all times 
3) I've enabled scrolling for POI and Vector popups when there are too many fields to display within the map canvas
4) I fixed a bug, where the popup didn't close on mouseout

I've manually checked the Aggregation and Visualization popup types and all seems ok with this change:
![POIPopupWorkingWithMANYfields](https://user-images.githubusercontent.com/36197976/72250137-d29bb900-35f2-11ea-8b2a-da3cbc1217ab.gif)

szydan commented on public/backwardsCompatible.js:
```diff
@@ -1,4 +1,5 @@
 import _ from 'lodash';
+import uuid from 'uuid';
```
not used remove
